File: college-placement-portal/backend/scripts/convert_to_sqlite.py
import re
import sys

with open('prisma/schema.prisma', 'r') as f:
    content = f.read()

# 1. Change provider to sqlite and url to file:./dev.db
content = re.sub(r'provider = "postgresql"', 'provider = "sqlite"', content)
content = re.sub(r'url      = env\("DATABASE_URL"\)', 'url      = "file:./dev.db"', content)

# 2. Find all enums to replace them in models
enum_names = re.findall(r'enum (\w+) \{', content)
print(f"Found enums: {enum_names}")

# 3. Comment out enum blocks
for enum_name in enum_names:
    pattern = rf'enum {enum_name} \{{[\s\S]*?\}}'
    content = re.sub(pattern, lambda m: f"// {m.group(0).replace('\n', '\n// ')}", content)

# 4. Replace enum types in models with String and update @default(...)
for enum_name in enum_names:
    # Match: fieldName  EnumName  @default(VALUE)
    # Group 1: whitespace before EnumName, Group 2: EnumName, Group 3: whitespace after, Group 4: @default(VALUE) or nothing
    # This is tricky due to Prisma's optional defaults and attributes.
    # We'll do a simpler replacement for now and fix if needed.
    content = re.sub(rf'\b{enum_name}\b', 'String', content)

# 5. Fix @db.Text (SQLite doesn't need it or use it as TEXT)
content = content.replace('@db.Text', '')

with open('prisma/schema.sqlite.prisma', 'w') as f:
    f.write(content)

print("Created prisma/schema.sqlite.prisma")

File: college-placement-portal/backend/scripts/convert_to_sqlite.js
const fs = require('fs');
const path = require('path');

const schemaPath = path.join(__dirname, '..', 'prisma', 'schema.prisma');
const outPath = path.join(__dirname, '..', 'prisma', 'schema.sqlite.prisma');

let content = fs.readFileSync(schemaPath, 'utf8');

content = content.replace(/provider = "postgresql"/, 'provider = "sqlite"');
content = content.replace(/url\s+=\s+env\("DATABASE_URL"\)/, 'url = "file:./dev.db"');

const enumRegex = /enum\s+(\w+)\s+\{/g;
const enumNames = [];
let match;
while ((match = enumRegex.exec(content)) !== null) {
    enumNames.push(match[1]);
}

enumNames.forEach(name => {
    const regex = new RegExp(`enum\\s+${name}\\s+\\{[\\s\\S]*?\\}`, 'g');
    content = content.replace(regex, (m) => {
        return m.split('\n').map(line => `// ${line}`).join('\n');
    });
});

enumNames.forEach(name => {
    const regex = new RegExp(`\\b${name}\\b`, 'g');
    content = content.replace(regex, 'String');
});

content = content.replace(/@db\.Text/g, '');
content = content.replace(/\bJson\b/g, 'String');
content = content.replace(/@default\((?!(true|false|\d+(\.\d+)?)\b)(\w+)\)/g, '@default("$1")');

fs.writeFileSync(outPath, content);
console.log('Created schema.sqlite.prisma');
